### PR TITLE
add -x flag to enable cc flags

### PR DESCRIPTION
**Pull request by mattn**
This is suggestion based pull-request. I want to compile code with c11 features.

See https://github.com/cznic/99c/issues/2

So I added flag `-x` to enable some flags of cc.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -87,6 +153,8 @@ func (a *args) getopt(args []string) {
 			a.W = arg[2:]
 		case arg == "-c":
 			a.c = true
+		case arg == "-x":
```
Comment (cznic) [inline main.go]: Please find a flag name that is not used by gcc, it uses -x for settin the language of the next argument. That's why I started to use "silly" flag names like '-99lib'. The long term goal is to be able to do somehting like `CC=99c ./configure && make`, so we must avoid using existing gcc flags.

File: main.go [hunk 2]
```diff
@@ -87,6 +153,8 @@ func (a *args) getopt(args []string) {
...
+			a.opts = append(a.opts, a.extra(arg[2:]))
```
Comment (cznic) [inline main.go]: I think we need to replace this line by

```go
if i+1 >= len(args) {
	exit(2, "missing -x argument") // Note the name of '-x' has yet to be detrmined
}

a.opts = append(a.opts, a.extra(arg[i+1]))
arg[i+1] = ""
```

**Review by cznic — CHANGES_REQUESTED** [review]
Thanks for the PR. I like the approach of doing all the cc options at once in one PR very much, kudos for that!

Please see the inliine comments for requested changes, thank you.

File: main.go [hunk 3]
```diff
@@ -87,6 +153,13 @@ func (a *args) getopt(args []string) {
 			a.W = arg[2:]
 		case arg == "-c":
 			a.c = true
+		case arg == "-99extra":
+			if i+1 >= len(args) {
+				exit(2, "missing -99flag argument")
```
Comment (cznic) [inline main.go]: s/flag/extra/

**Review by cznic — CHANGES_REQUESTED** [review]
Two last small changes please.

File: main.go [hunk 4]
```diff
@@ -126,6 +199,37 @@ func (a *args) getopt(args []string) {
     	Use the specified pathname, instead of the default a.out, for
     	the executable file produced. If the -o option is present with
     	-c or -E, the result is unspecified.
+  -xEXTRA
```
Comment (cznic) [inline main.go]: Somehow one comment got lost, sorry

s/-xEXTRA/-99extra argument/

**Review by cznic — CHANGES_REQUESTED** [review]
Update of my previous review.

**Review by cznic — APPROVED** [review]
LGTM, please merge to master branch, thank you.

**Comment by mattn** [comment]
Thank you

**Pull request merged**